include/vara/Feature/Feature.h

```diff
@@ -27,36 +27,53 @@ class Feature {
   using relationship_iterator = RelationshipListTy ::iterator;
 
 private:
-  string Name;
+  std::string Name;
+  llvm::SetVector<llvm::Value *> Values;
+  size_t ID;
+  bool Opt;
+  std::optional<Location> Loc;
   llvm::SmallVector<Feature *, 1> Parents;
   FeatureListType Children;
   FeatureListType Excludes;
   FeatureListType Implications;
   FeatureListType Alternatives;
   RelationshipListTy Relationships;
-  bool Opt;
-  std::optional<Location> Loc;
 
 protected:
-  Feature(string Name, bool Opt, std::optional<Location> Loc)
-      : Name(std::move(Name)), Opt(Opt), Loc(std::move(Loc)) {}
+  Feature(std::string Name, bool Opt, std::optional<Location> Loc)
+      : Name(std::move(Name)), ID(std::hash<std::string>{}(Name)), Opt(Opt),
+        Loc(std::move(Loc)) {}
 
 public:
+  Feature(const std::string &Name, llvm::Value *Val)
+      : Name(Name), ID(std::hash<std::string>{}(Name)), Opt(false) {
+    Values.insert(Val);
+  }
+
   Feature(const Feature &) = delete;
   Feature &operator=(const Feature &) = delete;
   virtual ~Feature() = default;
 
-  [[nodiscard]] string getName() const { return Name; }
+  [[nodiscard]] std::string getName() const { return Name; }
```

Comment:
Could be a string ref

Reply:
Will be resolved with #6 and merged afterwards